### PR TITLE
update Rust and c API for cancel_task

### PR DESCRIPTION
**Pull request by ya0guang**
## Description

- Update Rust and c API for cancel_task
- Add new tests for SDK
- Repaired an error in management service

Fixes # (issue)

## Type of change (select or add applied and delete the others)

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] API change with a documentation update
- [X] Additional test coverage
- [ ] Code cleanup or just sync with upstream third-party crates

## Checklist

- [X] Fork the repo and create your branch from `master`.
- [X] If you've added code that should be tested, add tests.
- [X] If you've changed APIs, update the documentation.
- [ ] Ensure the tests pass (see CI results).
- [ ] Make sure your code lints/format.

### CODE REVIEW

File: services/management/enclave/src/service.rs [hunk 1]
```diff
@@ -662,17 +662,24 @@ impl TeaclaveManagement for TeaclaveManagementService {
...
 
+                task.update_result(TaskResult::Err(TaskFailure {
+                    reason: "Task canceled".to_string(),
+                }))
+                .map_err(|_| TeaclaveManagementServiceError::PermissionDenied)?;
                 let ts: TaskState = task.into();
                 self.write_to_db(&ts)
                     .map_err(|_| TeaclaveManagementServiceError::StorageError)?;
+
+                log::warn!("Canceled Task writtenback");
```
Comment (bigdata-memory) [inline services/management/enclave/src/service.rs:682]: For the consistency, please adding `:` thanks.

Comment (bigdata-memory) [inline services/management/enclave/src/service.rs:682]: like `Canceled task: writtenback`

Comment (ya0guang) [reply services/management/enclave/src/service.rs]: Thanks! Just adde`:`

**Review by mssun — APPROVED** [review]
LGTM.

**Pull request merged**